### PR TITLE
Configure recommended JDBC connection-properties by default for PostgreSQL databases

### LINKED ISSUE #380: connection-settings parameter in dbconfig.xml template should be connection-properties instead
I guess this was a mental typo :-)

Will send a PR soon

### PR DESCRIPTION
**Pull request by oranenj**
For PostgreSQL only. Also fixes #380

### CODE REVIEW

**Comment by oranenj** [comment]
Reference: https://confluence.atlassian.com/jirakb/connection-problems-to-postgresql-result-in-stuck-threads-in-jira-1047534091.html

Installing the upgraded driver via the module would also be useful, but out of scope for this PR.

**Comment by oranenj** [comment]
For most installations not configuring the values does not matter, but JIRA will complain about it on upgrades so let's do it.

File: manifests/config.pp [hunk 1]
```diff
@@ -115,8 +115,13 @@
   $pool_test_while_idle = pick($jira::pool_test_while_idle, true)
   $pool_test_on_borrow = pick($jira::pool_test_on_borrow, false)
 
-  # This is just for consistency
-  $connection_settings = $jira::connection_settings
+  # JIRA will complain if these aren't set for PostgreSQL (will work fine though)
+  # https://confluence.atlassian.com/jirakb/connection-problems-to-postgresql-result-in-stuck-threads-in-jira-1047534091.html
+  if $jira::db == 'postgresql' {
```
Comment (bastelfreak) [inline manifests/config.pp:120]: personally, I prefer a selector statement, but if/else works as well.

File: templates/dbconfig.xml.epp [hunk 2]
```diff
@@ -51,7 +51,7 @@
     <time-between-eviction-runs-millis><%= $jira::config::time_between_eviction_runs %></time-between-eviction-runs-millis>
 <% } -%>
 <% if $jira::config::connection_settings { -%>
-    <connection-settings><%= $jira::config::connection_settings %></connection-settings>
+    <connection-properties><%= $jira::config::connection_settings %></connection-properties>
```
Comment (bastelfreak) [inline templates/dbconfig.xml.epp:54]: is that a backwards incompatible change?

Comment (oranenj) [reply templates/dbconfig.xml.epp:54]: No; the connection-settings option never did anything.

File: manifests/config.pp (hunk 1 above)
Comment (oranenj) [reply manifests/config.pp:120]: I prefer a simple if / else until there are more choices; I can never remember the selector syntax and it feels very clunky with only one option.

**Review by bastelfreak — APPROVED** [review]

**Comment by oranenj** [comment]
What's up with Ubuntu failing? I did nothing that would affect those tests...

**Comment by oranenj** [comment]
I'd like to merge this even though Ubuntu is failing. It seems there's just something in the `mysql` module that's not quite idempotent

**Comment by oranenj** [comment]
Bah, attempting to fix the tests just makes things worse.

**Comment by oranenj** [comment]
Since the only changes after approval were to make the MySQL tests run properly, I'll just merge this

**Pull request merged**